src/oncall/ui/static/js/oncall.js

```diff
@@ -2021,6 +2022,23 @@ var oncall = {
       }
     },
     renderPage: function(data){
+      $.when(
+        oncall.data.modesPromise
+      ).done(function() {
+        let contactModes = [];
+
+        for(let key in data.contacts)
+        {
+          let currentMode = oncall.data.modes.find(x => x.name === key);
+          contactModes.push({
+            label: currentMode.label,
+            mode: key,
+            value: data.contacts[key]
+          });
+        }
+        data.contactmodes = contactModes;
+        data.telmodes = ["call", "sms"];
```

Comment:
Can we move this to the top in settings.data, and reference it here with self.data.telmodes? this way the models are stored in one location.